### PR TITLE
Code to ensure return url domain belongs to hmcts.net/gov.uk

### PR DESCRIPTION
**Pull request by AyanDuttaHMCTS**
### JIRA link (if applicable) ###

Code to ensure return url domain belongs to hmcts.net/gov.uk

### Change description ###

Code to ensure return url domain belongs to hmcts.net/gov.uk

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/CardPaymentController.java [hunk 1]
```diff
@@ -232,4 +237,24 @@ public String return403(AccessDeniedException ex) {
         return ex.getMessage();
     }
 
+    private boolean validateReturnUrl(String returnUrl) throws URISyntaxException {
```
Comment (aravindan89) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/CardPaymentController.java:240]: Hi @AyanDuttaHMCTS - Can we move this out of controller if possible please?

Comment (AyanDuttaHMCTS) [reply api/src/main/java/uk/gov/hmcts/payment/api/controllers/CardPaymentController.java:240]: Good Catch, Let me check if we can move this to some util or validator class.

File: api/src/test/java/uk/gov/hmcts/payment/api/componenttests/CardPaymentControllerTest.java [hunk 2]
```diff
@@ -1025,6 +1025,68 @@ public void retrieveCardPaymentStatuses_byInvalidPaymentReferenceTest() throws E
...
+        result = restActions
+            .post("/card-payments", cardPaymentRequest)
+            .andExpect(status().isBadRequest())
+            .andReturn();
+
+        assertEquals("returnUrl: Must be an internal domain of hmcts.net or gov.uk", result.getResponse().getContentAsString());
+    }
+
+    @Test
+    public void createCardPayment_withVvalidReturnUrl_shouldReturn201Test() throws Exception {
```
Comment (YasharRahvar) [inline api/src/test/java/uk/gov/hmcts/payment/api/componenttests/CardPaymentControllerTest.java:1065]: spelling mistake here "withVvalid"

Comment (AyanDuttaHMCTS) [reply api/src/test/java/uk/gov/hmcts/payment/api/componenttests/CardPaymentControllerTest.java:1065]: :-) will correct, Thanks

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/CardPaymentController.java (hunk 1 above)
Comment (AyanDuttaHMCTS) [reply api/src/main/java/uk/gov/hmcts/payment/api/controllers/CardPaymentController.java]: Changes Done

File: api/src/test/java/uk/gov/hmcts/payment/api/componenttests/CardPaymentControllerTest.java (hunk 2 above)
Comment (AyanDuttaHMCTS) [reply api/src/test/java/uk/gov/hmcts/payment/api/componenttests/CardPaymentControllerTest.java]: Changes Done

**Review by aravindan89 — APPROVED** [review]

**Review by YasharRahvar — APPROVED** [review]

**Pull request closed without merging**